Sources/NostrSDK/Events/RelayListMetadataEvent.swift

```diff
@@ -0,0 +1,154 @@
+    }
+
+    init(tags: [Tag] = [], createdAt: Int64 = Int64(Date.now.timeIntervalSince1970), signedBy keypair: Keypair) throws {
+        try super.init(kind: .relayListMetadata, content: "", tags: tags, createdAt: createdAt, signedBy: keypair)
+    }
+
+    /// The list of ``UserRelayMetadata`` that describes preferred relays for discovering the user's content and receiving fresh content from others
+    public var relayMetadataList: [UserRelayMetadata] {
+        tags.compactMap { UserRelayMetadata(tag: $0) }
+    }
+}
+
+/// Describes a preferred relay for discovering a user's content and receiving fresh content from others.
+public struct UserRelayMetadata: Equatable {
+    /// The URL of the preferred relay.
+    public let relayURL: URL
+
+    /// The relay marker describing what type of events might be found from the preferred relay.
+    public let marker: Marker
+
+    public enum Marker {
+        /// When seeking events about the user who authored the ``RelayListMetadataEvent``, where the user was tagged,
+        /// clients SHOULD use this relay as a read relay.
+        case read
+
+        /// When seeking events from the user who authored the ``RelayListMetadataEvent``,
+        /// clients SHOULD use this relay as a write relay.
+        case write
+
+        /// When seeking events about the user who authored the ``RelayListMetadataEvent``, where the user was tagged,
+        /// or when seeking events from the user who authored the ``RelayListMetadataEvent``,
+        /// clients SHOULD use this relay as a read and write relay.
+        case readAndWrite
+    }
+
+    /// Creates a ``UserRelayMetadata`` from a ``Tag``.
+    /// The tag must have a tag name of `r`, value of a valid relay URL string, and, optionally, a marker of `read` or `write`.
+    /// If the marker is omitted, the relay is used for both read and write.
+    ///
+    /// A `nil` value is returned if the relay URL string is invalid or the marker is invalid.
+    ///
+    /// > Note: [NIP 65 - Relay List Metadata](https://github.com/nostr-protocol/nips/blob/master/65.md)
+    public init?(tag: Tag) {
+        guard tag.name == "r", let relayURL = try? RelayURLValidator.shared.validateRelayURLString(tag.value) else {
+            return nil
+        }
+
+        switch tag.otherParameters.first {
+        case "read":
+            marker = .read
+        case "write":
+            marker = .write
+        case .none:
+            marker = .readAndWrite
+        case .some:
+            return nil
+        }
+
+        self.relayURL = relayURL
+    }
+
+    /// Creates a ``UserRelayMetadata`` from a relay ``URL`` and ``Marker``.
+    ///
+    /// A `nil` value is returned if the relay URL string is invalid.
+    ///
+    /// > Note: [NIP 65 - Relay List Metadata](https://github.com/nostr-protocol/nips/blob/master/65.md)
+    public init?(relayURL: URL, marker: Marker) {
```

Comment:
Just a suggestion: What if you made the marker have a default value of `.readAndWrite`? That way callers in most cases wouldn't need to pass it at all.

```suggestion
    public init?(relayURL: URL, marker: Marker = .readAndWrite) {
```